### ISSUE TITLE
New Envelope Modes break MPE

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#6394]
Good news: The old modes all still work.

So here's the diagnosis on the new modes

**Poly:**
Press C, Press G, Slide C -> E and back, Slide G -> E and back.
In old and new mode you hear exactly what you expect.

Now set release longer

Press C, Press G, Release and Re-Press G within the envelope, Slide C->E->C and G->E->G
The C slides the G does not
Clearly that re-used voice has the wrong channel state for finding the pitch bend

**MONO (no st)**
Reset from zero, press C, press G, G->E->G hear it C->E->C don't, Release G, C->E->C hear it
New mode same gesture the G->E->G doesn't sound, the C->E->C bends the G, after the release it works
Clearly the stolen voice has the wrong channel state for finding the pitch bend. (A theme!)

Long release
C->E->C release but while voice is still ringing G->E->G release
Old mode you hear the double slide
New mode you hear the first but not the second. No slide returns until the note releases then the first gesture slides
Clearly the stolen voice has the wrong channel state for finding the pitch bend. (A theme!)

** MONO ST**
Reset from zero, press C, press G, G->E->G hear it C->E->C don't, Release G, C->E->C hear it
This works in both modes (because there is no new voice stealing in ST mode)

The bug here is C->E->C release, while ringing C->E-C and there is no bend
But *subseuent* legato notes get bend.
Basically the ST legato MPE works but the voice stealing on release doesnt
Clearly the stolen voice has the wrong channel state for finding the pitch bend. (A theme!)
